VoxelSniperCore/src/main/java/com/github/kevindagame/util/VersionChecker.kt

```diff
@@ -0,0 +1,122 @@
+package com.github.kevindagame.util
+
+import com.google.gson.Gson
+import java.io.InputStreamReader
+import java.net.HttpURLConnection
+import java.net.URL
+
+/**
+ * @author KevinDaGame
+ * A class that checks for the latest version of VoxelSniper-Reimagined on GitHub.
+ */
+class VersionChecker() {
+    private val releasesApiUrl = "https://api.github.com/repos/$REPOSITORY_OWNER/$REPOSITORY_NAME/releases"
+
+    /**
+     * Gets the latest version info from the GitHub API.
+     * @param currentVersion The current version of VoxelSniper-Reimagined.
+     *
+     * @return The latest version info, or null if the current version is the latest.
+     */
+    fun getLatestVersionInfo(currentVersion: String): VersionInfo? {
+        val releases = getReleasesFromApi()
+        if (releases.isEmpty()) {
+            return null
+        }
+        val latestRelease = releases[0]
```

Comment:
Could we assume the first one is the latest version?

Reply:
Good point, The api docs don't explicitly state this, though it seems to be based on publish date, this did make me think of something though, prereleases shouldn't trigger the update warning